integration/ingester_sharding_test.go

```diff
@@ -77,6 +80,11 @@ func TestIngesterSharding(t *testing.T) {
 				labels.MustNewMatcher(labels.MatchEqual, "name", "ingester"),
 				labels.MustNewMatcher(labels.MatchEqual, "state", "ACTIVE"))))
 
+			// Yes, we're sleeping in this test. We need to make sure that the ingesters have been part
+			// of the ring long enough before writing metrics to them to ensure that only the shuffle
+			// sharded ingesters will be queried for them when we go to verify the series written.
+			time.Sleep(queryIngestersWithinSecs * time.Second)
```

Comment:
The registration timestamp has second precision. I think it's safe to sleep for `(queryIngestersWithinSecs + 1)` seconds in order to avoid any flakyness due to rounding issues.